Testscripts/Linux/XDP-Action.sh

```diff
@@ -0,0 +1,291 @@
++		return ping_reply(ctx);
++
+ 	return XDP_PASS;
+ }
+
+--
+2.17.1
+
+
+EOF
+	fi
+	if [[ ! -f /root/xdpdump_kern_"${ACTION}".patch ]]; then
+		LogErr "patch file '/root/xdpdump_kern_${ACTION}.patch' could not be created"
+		SetTestStateAborted
+		exit 1
+	fi
+	LogMsg "patch file for ${ACTION} is created successfully"
+}
+
+# Helper function
+# This function will start xdpdump on client 
+# and ping from server on eth1 network interface
+function ping_test () {
+	local logSuffix=$1
+
+	xdpdumpCommand="cd bpf-samples/xdpdump && timeout 20 ./xdpdump -i ${nicName} > ~/xdpdumpout_${logSuffix}.txt"
+	LogMsg "Starting xdpdump on client with command: ${xdpdumpCommand}"
+	ssh ${client} "sh -c '${xdpdumpCommand} &'"
+
+	LogMsg "Starting ping on server with command: ping -I ${nicName} -c 10 ${clientSecondIP} > ~/pingOut_${logSuffix}.txt"
+	ssh ${server} "ping -I ${nicName} -c 10 ${clientSecondIP} > ~/pingOut_${logSuffix}.txt"
+}
+
+# Helper function
+# This function will start tcpdump on client and start ping_test
+function ping_with_tcpdump () {
+	local logSuffix=$1
+	LogMsg "Ips client $client server $server"
+	# Start tcpdump
+	LogMsg "Starting tcpdump on $client"
+	ssh ${client} "sh -c 'timeout 10 tcpdump -i eth1 icmp -w tcpdumptest_${logSuffix}.pcap &'"
+	ping_test $logSuffix
+}
+
+# This function will remove patch from xdpdump
+function remove_xdpdump_patch() {
+	ssh ${client} "cd bpf-samples && patch -p1 -R < /root/xdpdump_kern_${ACTION}.patch"
+	check_exit_status "Reverting XDPDump patch: /root/xdpdump_kern_${ACTION}.patch"
+}
+
+
+UTIL_FILE="./utils.sh"
+
+# Source utils.sh
+. ${UTIL_FILE} || {
+		echo "ERROR: unable to source ${UTIL_FILE}!"
+		echo "TestAborted" > state.txt
+		exit 0
+}
+
+# Source constants file and initialize most common variables
+UtilsInit
+
+# Script start from here
+LogMsg "*********INFO: Script Execution Started********"
+
+# check for client and server ips are present
+for ip in ${client} ${server}
+do
+	CheckIP ${ip}
+	if [ $? -eq 1 ]; then
+		LogErr "ERROR: Please provide valide client and server ip. Invalide ip: ${ip}"
+		SetTestStateAborted
+		exit 1
+	fi
+done
+
+if [ "${ACTION}" == "DROP" ];then
+	LogMsg "Initializing validation for XDP Action ${ACTION}"
+	# Create patch file
+	create_xdpdump_patch
+	# Ping test before changing action
+	ping_test "before"
+	check_exit_status "ping test before drop config"
+	# get XDP dump drop patch
+
+	LogMsg "Applying XDP Drop patch"
+	# XDP Drop configuration
+	ssh ${client} "cd bpf-samples && patch -p1 < /root/xdpdump_kern_${ACTION}.patch"
+	check_exit_status "Applying patch"
+
+	# Build xdpdump
+	ssh ${client} "cd bpf-samples/xdpdump && make"
+	check_exit_status "Building xdpdump with ip-drop patch"
+
+	# Ping test after changing action
+	ping_test "after"
+	if [ $? -ne 0 ];then
+		LogErr "Error occured while executing ping command. Please check pingOut_after.txt logs"
+		SetTestStateFailed
+		remove_xdpdump_patch
+		exit 1
+	fi
+
+	remove_xdpdump_patch
```

Comment:
because if you rerun tests in the same environment it will break as patch is already applied. 
And it would be a nice practice to keep environment closer to default environment.